framework/src/mfem/executioners/MFEMExecutioner.C

```diff
@@ -32,7 +32,9 @@ MFEMExecutioner::MFEMExecutioner(const InputParameters & parameters)
     _mfem_problem(dynamic_cast<MFEMProblem &>(feProblem())),
     _problem_data(_mfem_problem.getProblemData())
 {
-  _app.setMFEMDevice(getParam<std::string>("device"), Moose::PassKey<MFEMExecutioner>());
+  if (isParamValid("device"))
+    _app.setMFEMDevice(getParam<std::string>("device"), Moose::PassKey<MFEMExecutioner>());
```

Comment:
```suggestion
  if (const auto device = queryParam<std::string>("device"))
    _app.setMFEMDevice(*device, Moose::PassKey<MFEMExecutioner>());
```

beautiful new moose

Reply:
Yeah, or even `getParam<T>("name", default)`

Reply:
Unfortunately 

```
    _file_base(const auto base = queryParam<std::string>("file_base"),
               base ? *base : _app.getOutputFileBase(true)),
```
is not valid C++